### PR TITLE
fleet pp model dpo

### PR DESCRIPTION
**Pull request by wtmlon**
#### Before submitting

- [ ] Lint code. If there are lint issues, please format the code first.

```shell
# Install and register `pre-commit` in the project folder
pip install pre-commit && pre-commit install

# Process previous code files separately
pre-commit run --file XXXX.py
```

- [ ] Add test cases into `tests` folder. If there are codecov issues, please add tests cases first.

### CODE REVIEW

File: paddleformers/transformers/gpt_provider.py [hunk 1]
```diff
@@ -165,7 +165,7 @@ def provide(self, pre_process=None, post_process=None, vp_stage=None) -> GPTMode
         )
         if vp_size and not is_flexible_pp_layout:
             assert (
-                self.num_layers // pp_size
+                self.num_hidden_layers // p_size
```
Comment (nepeplwu) [inline paddleformers/transformers/gpt_provider.py]: p_size？

Comment (wtmlon) [reply paddleformers/transformers/gpt_provider.py]: update

File: paddleformers/cli/hparams/finetuning_args.py [hunk 2]
```diff
@@ -462,7 +462,7 @@ def __post_init__(self):
 
         self.max_gradient_accumulation_steps = self.gradient_accumulation_steps
 
-        if self.pipeline_model_parallel_size > 1:
+        if self.HAS_PADDLEFLEET or self.pipeline_model_parallel_size > 1:
```
Comment (nepeplwu) [inline paddleformers/cli/hparams/finetuning_args.py]: self.HAS_PADDLEFLEET是否应该换成一个通用的工具类

File: paddleformers/transformers/glm4_moe/modeling.py [hunk 3]
```diff
@@ -1686,9 +1696,19 @@ class Glm4MoeForCausalLMPipeFleet(Glm4MoePreTrainedModel, GeneralModelForCausalL
     is_fleet = True
 
     def __new__(cls, config):
+        # Hybrid parallel config convert.
+        config.tensor_model_parallel_size = max(config.tensor_model_parallel_size, 1)
```
Comment (nepeplwu) [inline paddleformers/transformers/glm4_moe/modeling.py:1771]: 解释下这里的做法原因

File: paddleformers/transformers/configuration_utils.py [hunk 4]
```diff
@@ -376,7 +377,7 @@ class LlmMetaConfig:
         (
             "moe_router_force_load_balancing",
             bool,
-            True,
+            False,
```
Comment (nepeplwu) [inline paddleformers/transformers/configuration_utils.py:367]: 默认值为什么要修改

File: paddleformers/transformers/glm4_moe/modeling.py (hunk 3 above)
Comment (wtmlon) [reply paddleformers/transformers/glm4_moe/modeling.py:1771]: 默认值问题，fomers这边默认值是-1，用到fleet那边会报错

File: paddleformers/transformers/configuration_utils.py (hunk 4 above)
Comment (wtmlon) [reply paddleformers/transformers/configuration_utils.py:367]: 这个是fakegate，就是用噪声模拟gate logits，我理解正常组网不需要？https://github.com/PaddlePaddle/PaddleFleet/blob/e7982110a844d90e6958d38036a71ba105b1999e/src/paddlefleet/transformer/moe/moe_router.py#L69

File: paddleformers/trainer/trainer.py [hunk 5]
```diff
@@ -48,6 +48,8 @@
 from paddle.distributed.auto_parallel._utils import _patch_grads_for_step
 from paddle.distributed.fleet.meta_parallel import PipelineLayer
 
+from paddleformers.transformers.gpt_provider import GPTModel
```
Comment (lugimzzz) [inline paddleformers/trainer/trainer.py]: 得判断一下is_paddlefleet_available，要不会报错

File: paddleformers/cli/train/dpo/dpo_trainer.py [hunk 6]
```diff
@@ -20,8 +20,16 @@
 from paddleformers.nn.criterion import CriterionLayer
 from paddleformers.peft.lora.lora_model import AVAILABLE_LAYERS
 from paddleformers.trainer import Trainer
+from paddleformers.transformers.gpt_provider import GPTModel
```
Comment (lugimzzz) [inline paddleformers/cli/train/dpo/dpo_trainer.py]: 和trainer的问题一样

File: paddleformers/trainer/trainer.py (hunk 5 above)
Comment (wtmlon) [reply paddleformers/trainer/trainer.py]: done

File: paddleformers/cli/train/dpo/dpo_trainer.py (hunk 6 above)
Comment (wtmlon) [reply paddleformers/cli/train/dpo/dpo_trainer.py]: done

**Review by lugimzzz — APPROVED** [review]
LGTM
